modules/tile-converter/src/3d-tiles-converter/helpers/texture-atlas.ts

```diff
@@ -1,12 +1,14 @@
+import {TypedArray} from '@loaders.gl/loader-utils';
+
 /**
  * Apply uvRegions to texture coordinates.
  * Spec - https://github.com/Esri/i3s-spec/blob/master/docs/1.7/geometryUVRegion.cmn.md
  * Shader formula vec2 uv = fract(texCoords) * (uvRegions.zw - uvRegions.xy) + uvRegions.xy;
  * @param texCoords
  * @param uvRegions
  */
-export function convertTextureAtlas(texCoords: Float32Array, uvRegions: Uint16Array): Float32Array {
-  const convertedTexCoords = new Float32Array(texCoords.length);
+export function convertTextureAtlas(texCoords: TypedArray, uvRegions: TypedArray): Float32Array {
+  const convertedTexCoords = new texCoords[Symbol.species](texCoords.length);
```

Comment:
Hmm, I have never needed to use `Symbol.species`. Do we need this or can we get away with a simpler constructor. This feature seems to be for overriding default constructor when subclassing built in types? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/species. 

Also do we still need the original texCoords after conversion? If not consider converting "in place" and avoid an allocation?

Finally an unrelated comment: In the algorithm below there are lots of little arrays being created for each iteration. I would recommend creating a few scratch arrays before the loop and reusing those, to reduce memory pressure and improve performance.